docs/upgrade_guide.md

```diff
@@ -0,0 +1,30 @@
+# Upgrade Guide
+
+This guide describes how to upgrade Open Service Mesh (OSM).
+
+## Prerequisites
+- Kubernetes cluster with OSM installed
+- The [helm 3 CLI](https://helm.sh/docs/intro/install/)
+
+## Upgrade OSM using Helm
+Use the `helm` CLI to upgrade the OSM control plane from a Kubernetes cluster.
+
+Run `helm upgrade`. 
+```console
+# Add steps for helm upgrade
+```
+
+Run `helm upgrade --help` for more options.
+
+## CRD Upgrades
+Please check the `CRD Updates` section of the [release notes](https://github.com/openservicemesh/osm/releases) to see if additional steps are required to update the CRDs used by OSM. If the new release does contain updates to the CRDs, it is required to first delete existing CRDs and the associated Custom Resources prior to upgrading.
+
+In the `./scripts/cleanup` directory we have included a helper script to delete those CRDs and Custom Resources: `./scripts/osm-cleanup.sh`
+
+## Upgrade support
+Upgrades are supported between consecutive minor versions. Skipping minor versions is not supported.
```

Comment:
I'm unclear on what supported means here since we don't yet have a CLI command that does the upgrade. Let's talk more about this. Will think about it for a bit.

Reply:
Yes I'd love to generate more of a discussion around this. To me, support means that we have validated that between minor releases, a helm upgrade will work smoothly (barring the CRD stuff). 

But I also believe we are kind of still in an "alpha" stage so can we really guarantee that? I'm not sure how to anticipate what kinds of change would result in a breaking release

Reply:
I've removed this section for now. We can add it back once we've had more time to discuss and when we add the osm cli upgrade command

Reply:
sounds good